### PR TITLE
Add peer assessment hint view

### LINKED ISSUE #622: Native view for peer assessments
**Is your feature request related to a problem? Please describe.**
To improve the offline (and app) experience, all course items should have a native view representation. Peer assessments are unlikely to supported by the native apps in the near future.

**Describe the solution you'd like**
The app should show an overview of the important data (title, type, instructions) and a note that this part is not optimized for the app. The user then knowingly decides to open the external page by pressing a button.

**Additional context**
Similar to #581

### PR DESCRIPTION
**Pull request by caterinamandel98**
Fixes #622

### CODE REVIEW

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        formatter.dateFormat = "dd. MMM yyyy"
+
+        if let deadline = self.courseItem?.deadline {
+            deadlineExpired = deadline.inPast
+            self.deadlineDateView.isHidden = false
+            self.deadlineLabel.text = Self.dateFormatter.string(from: deadline)
+        }
+
+        self.deadlineMessageView.isHidden = !deadlineExpired
+        self.peerAssessmentInfoView.isHidden = deadlineExpired
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: This code should be moved the `updateView()`

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        formatter.minimumFractionDigits = 1
+        formatter.maximumFractionDigits = 1
+        return formatter
+    }()
+
+    private static let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd. MMM yyyy"
+        return formatter
+    }()
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: Unfortunately, this date format is not localized. Additionally, this does not include the time and time zone, which are important for the learner when dealing with deadlines ;) Have a look at `\CourseData.formattedDateWithTimeZone`. You can reuse this code to achieve a uniform deadline format in the app.

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        super.viewDidLoad()
+
+        var deadlineExpired = false
+
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd. MMM yyyy"
+
+        if let deadline = self.courseItem?.deadline {
+            deadlineExpired = deadline.inPast
+            self.deadlineDateView.isHidden = false
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: I sort of dislike this approach. You rely on the initial value from the storyboard file. However, this value is hidden to the developer only looking at this file. Please make sure to also cover the opposite case (hiding the view), so the state is independent of the storyboard.

I would suggest something like this:
```swift
let deadlineExpired = self.courseItem?.deadline?.inPast ?? false
self.deadlineDateView.isHidden = self.courseItem?.deadline == nil
```

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        switch peerAssessment.type {
+        case "team":
+            self.peerAssessmentTypeLabel.text = "Team Peer Assessment"
+            self.teamAssessmentView.isHidden = false
+            self.soloAssessmentView.isHidden = true
+        default:
+            self.peerAssessmentTypeLabel.text = "Peer Assessment"
+            self.teamAssessmentView.isHidden = true
+            self.soloAssessmentView.isHidden = false
+        }
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: As we discussed in the chat, you can simply use a single view for `soloAssessmentView` and `teamAssessmentView`.

Don't forget to use `NSLocalizedString` here, even if the value are the same for now.

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                self?.assessmentInstructionsView.attributedText = attributedString
+            }
+        }
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard let item = self.courseItem else { return }
+        if let typedInfo = R.segue.iOSPeerAssessmentViewController.openPeerAssessmentURL(segue: segue) {
+            typedInfo.destination.url = item.url
+               }
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: Wrong indentation ☹️

File: iOS/ViewControllers/Courses/PeerAssessmentViewController.swift [hunk 6]
```diff
@@ -0,0 +1,136 @@
+//
+//  Created for xikolo-ios under MIT license.
+//  Copyright © HPI. All rights reserved.
+//
+
+import Common
+import UIKit
+
+class PeerAssessmentViewController: UIViewController {
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentViewController.swift]: Similar to the `LTIHintViewController`, this view controller should be named `PeerAssessmentHintViewController` as we do not cover a peer assessment with this view but only a page before the peer assessment.

File: iOS/ViewControllers/Courses/PeerAssessmentHintViewController.swift [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            return nil
+        }
+
+        var dateText = Self.dateFormatter.string(from: date)
+        if let timeZoneAbbreviation = TimeZone.current.abbreviation() {
+            dateText += " (\(timeZoneAbbreviation))"
+        }
+
+        return dateText
+    }
```
Comment (mathebox) [inline iOS/ViewControllers/Courses/PeerAssessmentHintViewController.swift]: I did some more research. We can simply use `timeStyle: .long` to include to time zone - no manual string concatenation needed :)

Comment (caterinamandel98) [reply iOS/ViewControllers/Courses/PeerAssessmentHintViewController.swift]: Alright, thanks for the note, I'll adjust that:)

**Pull request merged**